meta-balena-common/conf/distro/include/balena-os.inc

```diff
@@ -27,20 +27,16 @@ python () {
 }
 
 def set_default_kernel_cmdline(d):
-    cmdline = 'console=null quiet splash vt.global_cursor_default=0 consoleblank=0'
-    if d.getVar('OS_DEVELOPMENT', True) is '1':
-        cmdline = ""
-        for serial_console in d.getVar('SERIAL_CONSOLES', True).split():
-            try:
-                baudrate,port = serial_console.split(';')
-                console = "console=" + port + "," + baudrate + "n8"
-                if len(cmdline) is 0:
-                    cmdline = console
-                else:
-                    cmdline = cmdline + " " + console
-            except:
-                return
-    return cmdline
+    prod_cmdline = 'console=null quiet splash vt.global_cursor_default=0 consoleblank=0'
+    serial_consoles = d.getVar('SERIAL_CONSOLES', True).split()
+    devmode_extra_args = ['verbose=true', 'debug=true']
+    devmode_cmdline = ' '.join([
+        'console={},{}s8'.format(port, baudrate)
+        for (baudrate,port) in (console.split(';') for console in serial_consoles)
+    ] + devmode_extra_args)
+    return (devmode_cmdline
```

Comment:
@jakogut is this really required? Making init scripts run in debug mode will change timings between dev and prod modes. I have personally never found this lacking as it's fairly easy to add kernel command line arguments on a device or a specific build.
Could you add some other OS people to the review list see what they think?

Reply:
I'll mark this as a draft awaiting further eyeballs.

Reply:
> As far as timing goes, I'm not sure how much impact this would have on development builds

Printing to a serial console is slow - timing bugs in the init script are not uncommon - I recall Michal's active partition mounting on secure boot issue for example. The risk not worth the conveniennce imo.